NSS-Therapeutic-Raccoons/BangazonSite
# Mh 35 add poduct to shopping cart

## Link to Ticket
Closes # [Ticket 1](https://github.com/NSS-Therapeutic-Raccoons/BangazonSite/issues/1) [Ticket 35](https://github.com/NSS-Therapeutic-Raccoons/BangazonSite/issues/35)

## Description of Proposed Changes
- Added ability to redirect an unregistered user to the login/registration page.

- Added ability for a logged in user to view the items in their current order

- Added ability to show the user a message if they try to view their order when they have no items in it

- Added ability for a user to click links to product details if they click the item name in the current order details

- Added ability to order to the products details page that will add the product to a user's current open order

- Added a button onto the product details page called _Add to Order_

- Added ability to create a new order for a user if they have no open orders

- Added ability to decrease the quantity of product in the database if the user adds the item to their order

- Added ability to show the user a message instead of the add to order button on the products details page if the quantity of the product is 0

- Shortened the list of displayed information on the products description page in order to comply with ticket 1's requirements of only showing the title, description, quantity, and price

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout <MH-35-AddPoductToShoppingCart>
```
1. Login to the admin account, which has the email: admin@admin.com and the password: Admin8*

2. Click on the _View Cart_ button in the navigation bar. You should see two items in the cart, a kite and a microwave 4.

3. Click on the name of either the kite or microwave 4 and it should take you to the details page for that product.

4. Click on the _Add to Order_ button and it will take you back to the Order details page, where you will see the newly added item at the bottom of the list.

5. Click on the same item as before and you should see that the quantity of the item has now changed to be 1 less than it was before you added it to the order. You can repeat this process for any number of items you wish until the quantity reaches zero. At that point the _Add to Order_ button will no longer be on the product details page and a message will appear instead to inform the user that the item is no longer in stock.

6.  Log out of the admin account and register a new user.

7. Clicking on the _View Cart_ button in the navigation bar will show a message informing the user that there are no items in their cart.

8. Clicking on the _Add to Order_ button on any product details page will create a new order for that user and will show the newly added item in the user's order.


## Impacted Areas in Application

List general components of the application that this PR will affect:

- Controllers/OrdersController.cs
- Controllers/ProductsController.cs
- Views/Orders/Details.cshtml
- Views/Products/Details.cshtml
- Views/Shared/_Layout.cshtml

## Mentions @username

@figamus
@ricky-bruner
@markhale94
@taylorlgulley
@askingalot

## Definition of Done

1. The project must be fully documented. This includes the following:
	* Comments for classes and methods added
	* Comments for models, controllers, and views added
2. The testing steps must completely work per ticket requirements.
3. Andy must smile approvingly and smirk with satisfaction following the smile.
